mldsa/fips202/fips202.c

```diff
@@ -559,15 +575,30 @@ __contract__(
  *              - uint64_t *s: pointer to input/output Keccak state
  *              - unsigned int r: rate in bytes (e.g., 168 for SHAKE128)
  **************************************************/
-static void keccak_squeezeblocks(uint8_t *out, size_t nblocks, uint64_t s[25],
-                                 unsigned int r)
+static void keccak_squeezeblocks(uint8_t *out, size_t nblocks,
+                                 uint64_t s[MLD_KECCAK_LANES], unsigned int r)
+__contract__(
+  requires(r < sizeof(uint64_t) * MLD_KECCAK_LANES)
+  requires(nblocks <= 8 /* somewhat arbitrary bound */)
```

Comment:
This bound should not be required.

Reply:
Actually... the bound should probably be set such that (r * nblocks) cannot be larger than UINT32_MAX (or should it be the largest value of size_t ??) e.g.
```
requires(nblocks < (UINT32_MAX / r))
``